z390test/src/main/groovy/org/z390/test/z390Test.groovy

```diff
@@ -0,0 +1,266 @@
+class z390Test {
+
+    var project_root = pathJoin('..')
+    String stdout = ""
+    String stderr = ""
+    var env = [:]
+    var fileData = [:]
+    var message = "z390Test - Test framework for z390 project"
+    int cmdTimeMs = 100000
+    boolean printOutput = false
+
+    File tempDir = null
+
+    static void main(String[] args) {
+        println new z390Test().message
+    }
+
+    z390Test() {
+        /**
+         * class constructor
+         */
+        if (System.getenv('Z390_PROJECT_ROOT')) {
+            this.project_root = new File(System.getenv('Z390_PROJECT_ROOT'))
+        }
+        if (System.getenv('Z390_PRINT_OUTPUT')) {
+            printOutput = true
+        }
+    }
+    @BeforeEach
+    void setUp() {
+        this.stdout = ""
+        this.stderr = ""
+        this.fileData = [:]
+    }
+        @AfterEach
+        void cleanUp() {
+            if (this.tempDir)
+                this.tempDir.deleteDir()
+        }
+
+    def createTempFile(String fileName, String fileContents, boolean returnExt=true) {
+        /**
+         * Utility method for creating temp option files used in tests
+         */
+        if (!tempDir) {
+            this.tempDir = File.createTempDir()
+            this.tempDir.deleteOnExit()
+        }
+        var fullFileName = pathJoin(tempDir.absolutePath, fileName)
+        println("Creating temp source: ${fullFileName}")
+        // to allow files to reference themselves, include {{fullFileName}} in contents
+        fullFileName = fullFileName.replace("\\", "\\\\")  // support windows files
+        fileContents = fileContents.replaceAll(/\{\{fullFileName}}/, fullFileName)
+        new File(fullFileName).with {
+            createNewFile()
+            write(fileContents)
+        }
+        String ext = fullFileName.substring(fullFileName.lastIndexOf("."))
+        if (!returnExt || ext.toUpperCase() in ['.MLC', '.CBL'])
+            fullFileName = filenameWithoutExtension(fullFileName)
+        return fullFileName
+    }
+
+    String basePath(String... pathItems) {
+        var fullPathItems = [this.project_root, *pathItems]
+        return new File(fullPathItems.join(File.separator)).getCanonicalPath()
+    }
+
+    String basePathRelative(String... pathItems) {
+        var fullPathItems = [this.project_root, *pathItems]
+        return fullPathItems.join(File.separator)
+    }
+
+    static String filenameWithoutExtension(String filename) {
+        String result = filename[0..<filename.lastIndexOf('.')]
+        return result
+    }
+
+    def static pathJoin(String... pathItems) {
+        return pathItems.join(File.separator)
+    }
+
+    def getEnvList() {
+        def envList = []
+        this.env.each{item -> envList.add("${item.key}=${item.value}")}
+        return envList
+    }
+
+    def clean(String asmFileExcludingExtension) {
+        /**
+         * clear files for module
+         */
+        for (ext in ["PRN", "ERR", "OBJ", "390", "OBJ", "LOG", "LST", "STA"]) {
```

Comment:
The bash/asm script also deletes (trace) files with extension TR* (that is, TRA, TRE, etc).

Reply:
OK, I have added logic to the class to support deleting all TR* files as part of the clean process.